src/components/AuthStoreSetter.js

```diff
@@ -0,0 +1,28 @@
+import _ from 'lodash/fp'
+import { useEffect } from 'react'
+import { useAuth } from 'react-oidc-context'
+import { processUser, reloadAuthToken } from 'src/libs/auth'
+import { useOnMount } from 'src/libs/react-utils'
+import { authStore } from 'src/libs/state'
+
+
+const AuthStoreSetter = () => {
+  const auth = useAuth()
+
+  useOnMount(() => authStore.update(_.set(['authContext'], auth)))
+
+  useEffect(() => {
+    const cleanupFns = [
+      auth.events.addAccessTokenExpiring(reloadAuthToken),
+      auth.events.addUserLoaded(processUser),
+      auth.events.addUserUnloaded(processUser)
+    ]
+    return _.over(cleanupFns)
```

Comment:
As written, this will not return a function to invoke upon cleanup, but the results of running all of the `cleanupFns`. I believe the cleanup will be invoked when as soon as this is run, not when `auth` changes.

Side note: TIL about `_.over`

```suggestion
    return () => _.over(cleanupFns)
```